src/Integrations/Greenhouse/GreenhouseClient.php

```diff
@@ -135,15 +140,28 @@ public function postApplication(string $itemId, array $params, array $files, str
 			$this->prepareFiles($files)
 		);
 
-		$response = \wp_remote_post(
-			self::BASE_URL . "boards/{$this->getBoardToken()}/jobs/{$itemId}",
+		$filterName = Filters::getGeneralSettingsFilterName('httpRequestTimeout');
+
+		// Curl used because files are not sent via wp request.
+		$curl = \curl_init(); // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_init
+		\curl_setopt_array( // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_setopt_array
+			$curl,
 			[
-				'headers' => $this->getHeaders(true),
-				'body' => \wp_json_encode($body),
+				\CURLOPT_URL => self::BASE_URL . "boards/{$this->getBoardToken()}/jobs/{$itemId}",
+				\CURLOPT_HTTPAUTH => \CURLAUTH_BASIC,
+				\CURLOPT_RETURNTRANSFER => true,
+				\CURLOPT_TIMEOUT => \apply_filters($filterName, General::HTTP_REQUEST_TIMEOUT_DEFAULT),
+				\CURLOPT_POST => true,
+				\CURLOPT_POSTFIELDS => $body,
+				\CURLOPT_HTTPHEADER => $this->getHeaders(true),
 			]
 		);
+		$response = \curl_exec($curl); // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_exec
+		$code = \curl_getinfo($curl, \CURLINFO_RESPONSE_CODE); // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_getinfo
+
+		\curl_close($curl); // phpcs:ignore WordPress.WP.AlternativeFunctions.curl_curl_close
 
-		if (\is_wp_error($response)) {
+		if ($code === 401) {
```

Comment:
This error code doesn't match the returned one. It'd check for a range of 400 codes and just return the `$code` on line 174.